src/Microsoft.Azure.WebJobs.Host/Queues/Bindings/QueueBindingProvider.cs

```diff
@@ -187,7 +187,7 @@ private IAsyncCollector<IStorageQueueMessage> BuildFromQueueAttribute(QueueAttri
 
             internal IStorageQueue GetQueue(QueueAttribute attrResolved)
             {
-                var account = Task.Run(() => this._accountProvider.GetStorageAccountAsync(attrResolved, CancellationToken.None)).GetAwaiter().GetResult();
```

Comment:
Removing Task.Run is incorrect and can lead to deadlocks. 

---
In reply to: [122810016](https://github.com/Azure/azure-webjobs-sdk/pull/1190#discussion_r122810016) [](ancestors = 122810016)